subgraphs/_reference_/src/sdk/protocols/perpfutures/pool.ts

```diff
@@ -0,0 +1,810 @@
+      );
+      totalValueLockedUSD = totalValueLockedUSD.plus(amountUSD);
+    }
+
+    this.setTotalValueLocked(totalValueLockedUSD);
+  }
+
+  /**
+   * Updates the total value locked for this pool to the given value.
+   * Will also update the protocol's total value locked based on the change in this pool's.
+   */
+  setVolume(newVolume: BigDecimal): void {
+    const delta = newVolume.minus(this.pool.cumulativeVolumeUSD);
+    this.addVolume(delta);
+    this.save();
+  }
+
+  /**
+   * Adds the given delta to the total value locked for this pool.
+   * Will also update the protocol's total value locked based on the change in this pool's.
+   *
+   * @param delta The change in total value locked for this pool.
+   */
+  addVolume(delta: BigDecimal): void {
+    this.pool.cumulativeVolumeUSD = this.pool.cumulativeVolumeUSD.plus(delta);
+    this.save();
+
+    this.protocol.addVolume(delta);
+  }
+
+  /**
+   * Adds a given USD value to the pool and protocol supplySideRevenue. It can be a positive or negative amount.
+   * Same as for the rest of setters, this is mostly to be called internally by the library.
+   * But you can use it if you need to. It will also update the protocol's snapshots.
+   * @param rev {BigDecimal} The value to add to the protocol's supplySideRevenue.
+   */
+  addSupplySideRevenueUSD(rev: BigDecimal): void {
+    this.pool.cumulativeTotalRevenueUSD =
+      this.pool.cumulativeTotalRevenueUSD.plus(rev);
+    this.pool.cumulativeSupplySideRevenueUSD =
+      this.pool.cumulativeSupplySideRevenueUSD.plus(rev);
+    this.save();
+
+    this.protocol.addSupplySideRevenueUSD(rev);
+  }
+
+  /**
+   * Adds a given USD value to the pool and protocol protocolSideRevenue. It can be a positive or negative amount.
+   * Same as for the rest of setters, this is mostly to be called internally by the library.
+   * But you can use it if you need to. It will also update the protocol's snapshots.
+   * @param rev {BigDecimal} The value to add to the protocol's protocolSideRevenue.
+   */
+  addProtocolSideRevenueUSD(rev: BigDecimal): void {
+    this.pool.cumulativeTotalRevenueUSD =
+      this.pool.cumulativeTotalRevenueUSD.plus(rev);
+    this.pool.cumulativeProtocolSideRevenueUSD =
+      this.pool.cumulativeProtocolSideRevenueUSD.plus(rev);
+    this.save();
+
+    this.protocol.addProtocolSideRevenueUSD(rev);
+  }
+
+  /**
+   * Adds a given USD value to the pool and protocol's supplySideRevenue and protocolSideRevenue. It can be a positive or negative amount.
+   * Same as for the rest of setters, this is mostly to be called internally by the library.
+   * But you can use it if you need to. It will also update the protocol's snapshots.
+   * @param protocolSide {BigDecimal} The value to add to the protocol's protocolSideRevenue.
+   * @param supplySide {BigDecimal} The value to add to the protocol's supplySideRevenue.
+   */
+  addRevenueUSD(protocolSide: BigDecimal, supplySide: BigDecimal): void {
```

Comment:
Do we want this to be the function exposed to the user, then the functions that it calls are private and only handled by the sdk? In an attempt to reduce the number of options for the dev and make it simpler

Reply:
yeah, that makes sense. I will make the other two functions private. Let's say the user only wants to update the `protocolSideRevenue`, he can pass the other variable (`supplySideRevenue`) as zero. So exposing all three can be avoided which will make it simpler.